### PR TITLE
Add callbacks for statusDidChange and downloadDidProgress

### PR DESCRIPTION
**Pull request by cooperka**
See https://github.com/Microsoft/react-native-code-push#codepush for more details (scroll down a bit for the param descriptions). All I'm doing here is passing the options on to CodePush so it can handle them.

### CODE REVIEW

**Comment by cooperka** [comment]
Hi @geof90, feel free to merge this if you think it looks fine!

**Comment by cooperka** [comment]
@lostintangent and @duranmla, would you be able to merge this for me? It's been open for a while. Thanks!

**Comment by duranmla** [comment]
I'm not able to merge things here @cooperka not sure why you think that, on the other hand, I will suggest you to write some context here, to allow the person that is going to merge know what is the issue and how you solve it. regards! 💯

**Comment by geof90** [comment]
Sorry! Just saw this. Will review it later today.

**Comment by cooperka** [comment]
Sounds good @geof90! Let me know if you need more context besides the documentation that CodePush provides, and I'd be happy to go into more detail.

Sorry for the confusion @duranmla, I just saw that you had previously added https://github.com/geof90/react-native-code-push-saga/commit/7c962da57b7837f9fbbb1ccc5a58ea10f6adc6d9 so it appeared you had merge permissions 😊

**Comment by duranmla** [comment]
No worries mate @cooperka

File: README.md [hunk 1]
```diff
@@ -57,4 +57,13 @@ yield spawn(codePushSaga, {
...
-```
\ No newline at end of file
+
+// Invoke a callback every time the sync status changes
+// or show a status bar as the download progresses.
+// You can find a list of status types at
+// https://github.com/Microsoft/react-native-code-push#syncstatus.
+yield spawn(codePushSaga, {
+   statusDidChangeCallback: this.onSyncStatusChange,
+   downloadDidProgressCallback: this.onSyncDownloadProgress,
```
Comment (geof90) [inline README.md]: Can we make the event names match the ones in CodePush? so onSyncStatusChange, onSyncDownloadProgress, etc.

Comment (cooperka) [reply README.md]: Updated. I used the names Microsoft used in their docs, does that look ok?

**Comment by cooperka** [comment]
@geof90 ping :)

**Comment by geof90** [comment]
Thanks @cooperka !

**Pull request merged**